### PR TITLE
Use Bool by default for OneHotEncoding Transform

### LINKED ISSUE #7: One hot encoding transform
A one-hot-encoding transform for categorical variables

MWE
```julia
x = ["foo", "bar", "baz"]

ohe = OneHotEncoding()

Transform.apply(x, ohe)

# output
3×3 Array{Int64,2}:
 1  0  0
 0  1  0
 0  0  1
```

### PR DESCRIPTION
**Pull request by nicoleepp**
Closes #7

### CODE REVIEW

File: src/one_hot_encoding.jl [hunk 1]
```diff
@@ -6,35 +6,34 @@ One-hot encode the categorical value for each target element.
...
         # in the sparse matrix that results from applying the OneHotEncoding transform
         categories = Dict(value => i for (i, value) in enumerate(possible_values))
-        return new(categories)
+        return new{T}(categories)
     end
 end
 
 function _apply(x, encoding::OneHotEncoding; kwargs...)
     n_categories = length(encoding.categories)
+    results = zeros(Bool, length(x), n_categories)
```
Comment (rofinn) [inline src/one_hot_encoding.jl:32]: If we parameterized the `encoding` type by `R<:Real` then we could make this more flexible as `zeros(R, ...)`. The `true` below would just get converted on `setindex!`.

Comment (rofinn) [reply src/one_hot_encoding.jl]: Thanks, LGTM 👍🏻

File: src/one_hot_encoding.jl [hunk 2]
```diff
@@ -1,40 +1,44 @@
...
+function OneHotEncoding(possible_values::AbstractVector{T}) where T
+    return OneHotEncoding{Bool}(possible_values)
+end
 
-    results = zeros(Int, length(x), n_categories)
+function _apply(x, encoding::OneHotEncoding{R}; kwargs...) where R <: Real
+    n_categories = length(encoding.categories)
+    results = zeros(R, length(x), n_categories)
 
-    @views for (i, value) in enumerate(x)
```
Comment (bencottier) [inline src/one_hot_encoding.jl:35]: Did you remove `@views` just because it seems unnecessary? I don't understand it well myself.

**Review by bencottier — APPROVED** [review]
Looks good to me. I think just edit the title e.g. "Use Bool by default for OneHotEncoding Transform".

File: src/one_hot_encoding.jl [hunk 3]
```diff
@@ -1,40 +1,44 @@
...
+The element [i, j] is `true` if the i^th target in `x` corresponds to the j^th possible
+value and `false` otherwise. Note that `R`can be specified to determine the return type
+of results. It defaults to a `Matrix` of `Bool`s.
 
 Note that this Transform does not support specifying dims other than `:` (all dims) because
 it is a one-to-many transform (for example a `Vector` input produces a `Matrix` output).
 """
-struct OneHotEncoding <: Transform
-    categories::Dict{Any, Int}
+struct OneHotEncoding{R<:Real, T} <: Transform
```
Comment (glennmoy) [inline src/one_hot_encoding.jl:16]: do we need to parameterise this by `T` as well? It doesn't seem to be used for anything - and we don't mention it in the docstring.

**Review by glennmoy — APPROVED** [review]

File: src/one_hot_encoding.jl (hunk 3 above)
Comment (nicoleepp) [reply src/one_hot_encoding.jl:16]: It's used for space efficiency so we don't need to store a Dict of `Any`

**Pull request merged**